### PR TITLE
feat: Vitest setup

### PR DESCRIPTION
**Pull request by feliperodriguess**
- Setup Vitest (testing framework)
- Add example test
- Add some `Holograph` contract class tests
- Configure scripts

How to run the tests:

Make sure you have the following environment variables set in `packages/sdk/.env` file:

ETHEREUM_TESTNET_RPC=
ETHEREUM_MAINNET_RPC=
POLYGON_TESTNET_RPC=
POLYGON_MAINNET_RPC=

Then, run the command:

`cd packages/sdk && pnpm test`

Screenshot 2023-11-22 at 02 36 13

### CODE REVIEW

**Review by BravoNatalie — APPROVED** [review]
👍

File: packages/sdk/src/tests/contracts/Holograph.test.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  let config: Config
+  let providers: Providers
+
+  beforeAll(() => {
+    config = Config.getInstance(NETWORKS_MOCK)
+    providers = new Providers(config)
+  })
+
+  it('should be able to get the correct providers', () => {
+    const multiProviders = providers.providers
```
Comment (alexanderattar) [inline packages/sdk/src/tests/contracts/Holograph.test.ts:23]: `providers.providers` is kinda weird. might get confusing. maybe we can think about how to make it cleaner

**Review by alexanderattar — APPROVED** [review]
Nice this should help us have a simple structure for introducing and testing new classes

File: packages/sdk/src/tests/contracts/Holograph.test.ts (hunk 1 above)
Comment (feliperodriguess) [reply packages/sdk/src/tests/contracts/Holograph.test.ts]: just improved that

**Pull request merged**